src/debugger_requests.jl

```diff
@@ -19,10 +19,20 @@ function debug_notification(conn, state::DebuggerState, params::DebugArguments)
 
     put!(state.next_cmd, (cmd = :set_source_path, source_path = filename_to_debug))
 
-    ex = _parse_julia_file(filename_to_debug)
+    file_content = try
+        read(filename_to_debug, String)
+    catch err
+        # TODO Think about some way to return an error message in the UI
+        JSONRPC.send(conn, finished_notification_type, nothing)
+        put!(state.next_cmd, (cmd = :stop,))
+        return
+    end
+
+    ex = Base.parse_input_line(file_content; filename=filename_to_debug)
 
     # Empty file case
     if ex === nothing
+        # TODO Think about some way to return an error message in the UI
```

Comment:
Maybe we want to handle invalid expression case ?
i.e to check `ex.head === :incomplete` and `ex.head === :error`.

Reply:
Yes, good point! I think we did get crash reports about invalid code at some point. I don't remember whether the current version somewhat gracefully handles that or not, probably not... Maybe we wait until we get a crash report for it so that we see exactly where things go wrong? Or we could come up with our own repo case, of course. But totally agreed, we should handle that case as well.

Reply:
I just tried that on my machine and found [this line](https://github.com/julia-vscode/DebugAdapter.jl/blob/5eedc136ebf1c8c74780e393e66752dd254c1a4f/src/debugger_requests.jl#L44) results in an error (as expected, cause `JuliaInterpreter.prepare_thunk` doesn't expect invalid expression).